packages/frontend/src/redux/crossStateSelectors/selectCollectedAvailableForClaimData.js

```diff
@@ -0,0 +1,71 @@
+import BN from 'bn.js';
+import { isEmpty, some } from 'lodash';
+import { createSelector } from 'reselect';
+
+import { selectValidatorsFarmData } from '../slices/staking';
+import {
+    selectTokensFiatValueUSD,
+    selectTokenWhiteList,
+} from '../slices/tokenFiatValues';
+import { selectAllContractMetadata } from '../slices/tokens';
+
+const collectFarmingData = (args) => {
+    try {
+        if (some(args, isEmpty)) return [];
+
+        const {
+            validatorsFarmData,
+            contractMetadataByContractId,
+            tokenFiatValues,
+            tokenWhitelist,
+        } = args;
+        const filteredFarms = Object.keys(validatorsFarmData)
+            .reduce((acc, validatorId) => {
+                return [...acc, ...validatorsFarmData[validatorId].farmRewards];
+            }, [])
+            .filter((farm) => farm.active && +farm.balance > 0);
```

Comment:
If we move the filtering logic to the component, this whole block can simply be reduced to:
```suggestion
           const filteredFarms = Object.values(validatorsFarmData).reduce((acc, {farmRewards}) => [...acc, ...farmRewards], [])
```

Reply:
Done. Nice suggestion, I like it. Thank you!